### PR TITLE
Network Plugin - Disable routes

### PR DESCRIPTION
**Pull request by bizzguy**
## Summary

Mock routes may contain quite a bit of data for header values and response bodies.  Currently, to turn off a mock a user must delete the mock and then go through the process of creating it when needed again.  This change will allow a user to temporarily disable a mock without deleting it.

A checkbox on the Routes list is used to enable/disable mocks.  As shown in this screenshot:

image

## Changelog

Network Plugin - disable/enable mock routes

## Test Plan

Create mocks using sample app
Verify that mocks are working as expected
Disable the mocks
Verify that the client has been updated with mocks (minus the disabled ones) user Flipper Messages plugin
Use the sample app to send the disabled requests again and verify that they are not mocked

### CODE REVIEW

**Comment by mgoyanes** [comment]
@bizzguy 
Thanks for this. It's really helpful.

Tested and it is working as expected.

Found a minor bug.
image

If the route is selected, hovering on the checkbox shows an incorrect message. It should say "Uncheck to disable mock route".


I also have another suggestion. A button to disable/enable all routes at once. It may come in handy when we have a lot of routes.

File: desktop/plugins/network/index.tsx [hunk 1]
```diff
@@ -370,6 +373,19 @@ export function plugin(client: PluginClient<Events, Methods>) {
         }
         informClientMockChange(routes.get());
       },
+      enableRoute(id: string, routeChange: Partial<Route>) {
+        console.log(routeChange);
```
Comment (mweststrate) [inline desktop/plugins/network/index.tsx:377]: could you drop the `console.log` statements?

File: desktop/plugins/network/index.tsx [hunk 2]
```diff
@@ -370,6 +373,19 @@ export function plugin(client: PluginClient<Events, Methods>) {
...
+        console.log(routes.get()[id]);
+        routeChange.enabled = !routes.get()[id].enabled;
+        console.log(routeChange);
+        if (routes.get().hasOwnProperty(id)) {
```
Comment (mweststrate) [inline desktop/plugins/network/index.tsx:381]: NIT: `routes.get()[id]` should suffice

File: desktop/plugins/network/ManageMockResponsePanel.tsx [hunk 3]
```diff
@@ -248,19 +253,26 @@ export function ManageMockResponsePanel(props: Props) {
...
+                    onOk() {
+                      const nextId = getNextId(id);
+                      networkRouteManager.removeRoute(id);
+                      setSelectedId(nextId);
+                    },
+                    onCancel() {},
+                  });
+                },
+                (id) => {
+                  console.log('...enableRoute');
```
Comment (mweststrate) [inline desktop/plugins/network/ManageMockResponsePanel.tsx:272]: Please drop this one :)

File: desktop/plugins/network/index.tsx [hunk 4]
```diff
@@ -370,6 +373,19 @@ export function plugin(client: PluginClient<Events, Methods>) {
         }
         informClientMockChange(routes.get());
       },
+      enableRoute(id: string, routeChange: Partial<Route>) {
```
Comment (mweststrate) [inline desktop/plugins/network/index.tsx:376]: I think you can drop the whole `routeChange` argument, and simplify the entire update to (I found the current implementation a bit hard to follow with the empty object that swaps it's enabled attribute):

```
routes.update((draft) => {
   draft[id].enabled = !draft[id].enabled
});
```

**Review by mweststrate — CHANGES_REQUESTED** [review]
I think this is an awesome feature from usability perspective! Thanks for thinking about it. Left a few comments

**Comment by bizzguy** [comment]
> @bizzguy
> Thanks for this. It's really helpful.
> 
> Tested and it is working as expected.
> 
> Found a minor bug.
> image
> 
> If the route is selected, hovering on the checkbox shows an incorrect message. It should say "Uncheck to disable mock route".
> 
> I also have another suggestion. A button to disable/enable all routes at once. It may come in handy when we have a lot of routes.

I made the changes to the tool tip messages but did not add a disable/enable button.  I'm not sure how the UI for that should look so I'd like to think about it a bit more and create a PR for it later.

File: desktop/plugins/network/index.tsx (hunk 4 above)
Comment (bizzguy) [reply desktop/plugins/network/index.tsx]: I made the change to remove Partial.  However, I'm finding some situations where there is no "id" so I'd like to keep the other code for now (especially since it appears in some of the other existing methods).

Comment (bizzguy) [reply desktop/plugins/network/index.tsx]: The other code being:

`if (routes.get().hasOwnProperty(id)) {`

**Review by mweststrate — APPROVED** [review]
